### PR TITLE
Subscription UI vm broker

### PR DESCRIPTION
**Pull request by akhil-jha**
The first two tests are failing due to bz1948758. Third test is failing at Line 510: `session.organization.select(org_name=org.name)`. I checked with pdb, the host gets registered with the satellite successfully.

### CODE REVIEW

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Make sure to use single quote string wherever possible and I'll do a more technical check when I can.

File: tests/foreman/ui/test_subscription.py [hunk 1]
```diff
@@ -290,30 +289,28 @@ def test_positive_view_vdc_subscription_products(session):
...
+        activation_key=repos_collection.setup_content_data['activation_key']['name'],
+        install_katello_agent=False,
+    )
+    with session:
+        session.organization.select(org.name)
+        session.contenthost.add_subscription(rhel7_contenthost.hostname, VDC_SUBSCRIPTION_NAME)
+        provided_products = session.subscription.provided_products(VDC_SUBSCRIPTION_NAME)
+        # ensure that subscription provided products list is not empty and that the product is
+        # in the provided products.
+        assert provided_products and product_name in provided_products
```
Comment (JacobCallahan) [inline tests/foreman/ui/test_subscription.py:304]:
```suggestion
        assert product_name in provided_products
```
As long as provided_products is always a list (or other iterable), then the first check isn't needed.
This type of check is used throughout the other tests in this PR.

I know this was part of the original tests, but could you clean these up as well? :)

File: tests/foreman/ui/test_subscription.py [hunk 2]
```diff
@@ -510,22 +504,23 @@ def test_positive_candlepin_events_processed_by_STOMP(session):
...
-            assert "0 Failed" in response["message"]
+    rhel7_contenthost.install_katello_ca()
+    rhel7_contenthost.register_contenthost(org.name, ak.name)
+    with session:
+        session.organization.select(org_name=org.name)
+        host = session.contenthost.read(rhel7_contenthost.hostname, widget_names='details')[
+            'details'
+        ]
+        sub_status = host['subscription_status']
+        assert 'Unentitled' in sub_status
```
Comment (JacobCallahan) [inline tests/foreman/ui/test_subscription.py:515]:
```suggestion
        assert 'Unentitled' in host['subscription_status']
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/ui/test_subscription.py (hunk 2 above)
Comment (akhil-jha) [reply tests/foreman/ui/test_subscription.py]: Done

File: tests/foreman/ui/test_subscription.py (hunk 1 above)
Comment (akhil-jha) [reply tests/foreman/ui/test_subscription.py]: Done. :)

**Review by JacobCallahan — APPROVED** [review]
ACK, nice!

File: tests/foreman/ui/test_subscription.py [hunk 3]
```diff
@@ -250,7 +249,7 @@ def test_positive_access_manifest_as_another_admin_user(test_name):
 
 @pytest.mark.libvirt_content_host
```
Comment (jameerpathan111) [inline tests/foreman/ui/test_subscription.py:250]:
```suggestion
```

Comment (JacobCallahan) [reply tests/foreman/ui/test_subscription.py:250]: good catch!

**Review by jameerpathan111 — APPROVED** [review]

**Pull request merged**